Add 25, 2D, and 42 codes (#191)
neo4j/docs-status-codes: modules/ROOT/pages/errors/gql-errors.adoc

=== 42N83

Status description:: error: syntax error or access rule violation - impersonation disallowed while password change required. Cannot impersonate a user while password change required.

=== 42N84

Status description:: error: syntax error or access rule violation - TERMINATE TRANSACTION misses YIELD clause. WHERE clause without YIELD clause. Use 'TERMINATE TRANSACTION ... YIELD ... WHERE ...'.

=== 42N85

Status description:: error: syntax error or access rule violation - cannot specify both allowed and denied databases. Allowed and denied database options are mutually exclusive.

=== 42N88

Status description:: error: syntax error or access rule violation - cannot grant privilege. Permission cannot be granted for 'REMOVE IMMUTABLE PRIVILEGE'.

=== 42N89

Status description:: error: syntax error or access rule violation - invalid driver settings map. Failed evaluating the given driver settings. { $cause }

=== 42N90

Status description:: error: syntax error or access rule violation - cannot alter immutable composite database. Composite databases cannot be altered (database: { $db }).

=== 42I50

> Status description:: error: syntax error or access rule violation - token name too long. Invalid input { $input }. A { $tokenType } name cannot be longer than { $maxTokenLength }.

Reviewer: If the order should be alpha-numeric this one should be before 42N.. right?

Reviewer: Also in the implementation I can see that the message is 
`Invalid input { %s }... A { %s } name cannot be longer than { %s }.` Probably the '...' is a bug in the codebase and the message should be like you have written it here, but I will ask Emelie from kernel that implemented this one to confirm

Reviewer 2: The input that is printed is truncated and after discussions in the team-error channel on slack we said to add the three dots after to indicate that. We decided to truncate the input since the string is at least 16383 characters long and if we print the whole string it would be hard for the user to read the message. I think we must have forgotten to update the spreadsheet.

Reviewer: Thanks, I forgot about that discussion. Then we should update it here and in the spreadsheet to the ... format instead.

=== 2DN02

Status description:: error: invalid transaction termination - constituent commit failed. Failed to commit constituent transaction. See debug log for details.

=== 2DN03

Status description:: error: invalid transaction termination - transaction termination failed. Failed to terminate transaction. See debug log for details.

=== 2DN04

Status description:: error: invalid transaction termination - constituent transaction termination failed. Failed to terminate constituent transaction. See debug log for details.

=== 2DN05

Status description:: error: invalid transaction termination - failed to apply transaction. There was an error on applying the transaction. See logs for more information.

=== 2DN06

Status description:: error: invalid transaction termination - failed to append transaction. There was an error on appending the transaction. See logs for more information.

=== 2DN07

Status description:: error: invalid transaction termination - inner transactions still open. Unable to commit transaction because it still have non-closed inner transactions.

[[transaction-rollback]]
== Transaction rollback

> Transaction rollback errors occur when the transaction is rolled back due to an error, such as when the transaction rollback fails, or when the constituent rollback fails.

Author: Yeah, I see. How about this one? This group of errors is a bit self-explanatory. 
```suggestion
Transaction rollback errors occur when there is a failure in a transaction or a constituent transaction rollback.
```

Reviewer: Yeah that sounds clearer

=== 40N01

Status description:: error: transaction rollback - rollback failed. Failed to rollback transaction. See debug log for details.

=== 40N02

Status description:: error: transaction rollback - constituent rollback failed. Failed to rollback constituent transaction. See debug log for details.

[[syntax-error-access-rule-violation]]
== Syntax error or access rule violation

> Syntax error or access rule violation errors occur when there is a syntax error in the query, or when the query violates the access rules, such as when the query contains an invalid syntax, or when the query tries to access a database that does not exist, etc.

Reviewer: ```suggestion
Syntax error or access rule violation errors occur when a Cypher query contains invalid syntax or when a client request violates the access rules, such as when a query tries to access a database without enough privileges, etc.
```

I found your sentence here a bit repetitive, and also I am unsure about the database not exist example as that also occurs under another class. So I tried to reformulate it. Feel free to improve my English or to go with something in between our descriptions if you like.

=== 42I50

> Status description:: error: syntax error or access rule violation - token name too long. Invalid input `{ $input }`... A `{ $tokenType }` name cannot be longer than `{ $maxTokenLength }`.

Reviewer: ```suggestion
Status description:: error: syntax error or access rule violation - token name too long. Invalid input { $input }... A { $tokenType } name cannot be longer than { $maxTokenLength }.
```

Author: Ah, good catch! Thank you!

Author: Ah, yes, good catch!

[[transaction-rollback]]
== Transaction rollback

> Transaction rollback errors occur when the transaction is rolled back due to an error, such as when the transaction rollback fails, or when the constituent rollback fails.

Reviewer: This sentence is a little bit hard to understand. Maybe it is just my lacking English, but to me it sounds a bit like the transaction was rolled back because of an error and that error was that the rollback failed. That cannot be the intended meaning though, because that would be a circle dependency.
